scripts/migrations/2024_10_move_ean8_products_to_new_path.pl

```diff
@@ -0,0 +1,348 @@
+
+				if (    (!-e "$data_root/products/$path")
+					and (!-e "$www_root/images/products/$path"))
+				{
+					# File::Copy move() is intended to move files, not
+					# directories. It does work on directories if the
+					# source and target are on the same file system
+					# (in which case the directory is just renamed),
+					# but fails otherwise.
+					# An alternative is to use File::Copy::Recursive
+					# but then it will do a copy even if it is the same
+					# file system...
+					# Another option is to call the system mv command.
+
+					# In this script, we want to avoid creating copies as we are using zfs, so we use File::Copy move()
+
+					print STDERR ("moving product data $data_root/products/$old_path to $data_root/products/$path\n");
+					print $log ("moving product data $data_root/products/$old_path to $data_root/products/$path\n");
+
+					if (not move("$data_root/products/$old_path", "$data_root/products/$path")) {
+						print STDERR ("could not move product data: $!\n");
+						print $log ("could not move product data: $!\n");
+					}
+
+					print STDERR (
+						"moving product images $www_root/images/products/$old_path to $www_root/images/products/$path\n"
+					);
+
+					print $log (
+						"moving product images $www_root/images/products/$old_path to $www_root/images/products/$path\n"
+					);
+
+					if (not move("$www_root/images/products/$old_path", "$www_root/images/products/$path")) {
+						print STDERR ("could not move product images: $!\n");
+						print $log ("could not move product images: $!\n");
+					}
+
+					# If the code changed, need to update the product .sto file and to remove the old code from MongoDB and to add the new code in MongoDB
+					if ($new_code ne $code) {
+						my $product_ref = retrieve_product($product_id);
+						$product_ref->{code} = $new_code . '';
+						$product_ref->{id} = $product_ref->{code} . '';    # treat id as string;
+						$product_ref->{_id} = $product_ref->{code} . '';    # treat id as string;
+						store_product("fix-code-bot", $product_ref, "changed code from $code to $new_code");
+						$products_collection->delete_one({code => $code});
+						$products_collection->replace_one({"_id" => $product_ref->{_id}}, $product_ref, {upsert => 1});
+						print STDERR "updated code from $code to $new_code in .sto file and MongoDB\n";
+						print $log "updated code from $code to $new_code in .sto file and MongoDB\n";
+					}
+
+				}
+			}
+		}
+		else {
+			print STDERR "new path exist, not moving $old_path to $path\n";
+			print $log "new path exist, not moving $old_path to $path\n";
+			$not_moved++;
+		}
+
+		if ($new_code ne $code) {
+			$changed_code++;
+			print STDERR "changed code from $code to $new_code\n";
+			print $log "changed code from $code to $new_code\n";
+		}
+
+	}
+	else {
+		print STDERR "new $path is the same as old $old_path\n";
+		print $log "new $path is the same as old $old_path\n";
```

Comment:
do you really need to log it ?
It will make your log very verbose for no reason, IMO